engine/src/main/java/org/hibernate/validator/internal/engine/constraintvalidation/ConstraintTree.java

```diff
@@ -56,12 +60,11 @@
 
 	private final Type validatedValueType;
 
-	public ConstraintTree(ConstraintDescriptorImpl<A> descriptor, Type validatedValueType) {
-		this( descriptor, validatedValueType, null );
-	}
+	private volatile Optional<ConstraintValidator<A, ?>> constraintValidatorForDefaultConstraintValidatorFactory;
 
-	private ConstraintTree(ConstraintDescriptorImpl<A> descriptor, Type validatedValueType, ConstraintTree<?> parent) {
-		this.parent = parent;
+	private volatile ConcurrentMap<ConstraintValidatorFactory, Optional<ConstraintValidator<A, ?>>> constraintValidatorPerConstraintValidatorFactory;
```

Comment:
Similar to the `Optional` above, what's the reason for using it as map value here?

Reply:
OK, so after discussing with myself in the shower, I decided to go for the dummy constraint validator approach. Having a wrapper in all cases for a corner case is not a good idea.